src/html.rs

```diff
@@ -157,36 +157,51 @@ pub fn walk_and_embed_assets(
                     }
                 }
                 "img" => {
-                    for attr in attrs_mut.iter_mut() {
-                        if &attr.name.local == "src" {
-                            let value = attr.value.to_string();
-
-                            // Ignore images with empty source
-                            if value == EMPTY_STRING.clone() {
-                                continue;
-                            }
-
-                            if opt_no_images {
-                                attr.value.clear();
-                                attr.value.push_slice(TRANSPARENT_PIXEL);
-                            } else {
-                                let src_full_url: String =
-                                    resolve_url(&url, &value).unwrap_or(EMPTY_STRING.clone());
-                                let (img_dataurl, _) = retrieve_asset(
-                                    cache,
-                                    &src_full_url,
-                                    true,
-                                    "",
-                                    opt_user_agent,
-                                    opt_silent,
-                                    opt_insecure,
-                                )
-                                .unwrap_or((EMPTY_STRING.clone(), EMPTY_STRING.clone()));
-                                attr.value.clear();
-                                attr.value.push_slice(img_dataurl.as_str());
-                            }
+                    // Find source tags
+                    let mut found_src: Option<Attribute> = None;
+                    let mut found_datasrc: Option<Attribute> = None;
+                    let mut i = 0;
+                    while i < attrs_mut.len() {
+                        let name = attrs_mut[i].name.local.as_ref();
+                        if name.eq_ignore_ascii_case("src") {
+                            found_src = Some(attrs_mut.remove(i));
+                        } else if name.eq_ignore_ascii_case("data-src") {
+                            found_datasrc = Some(attrs_mut.remove(i));
+                        } else {
+                            i += 1;
                         }
                     }
+
+                    // If images are disabled, clear both sources
```

Comment:
I believe both src and data-src are clear at this point, we just decide to either set src="<transparent_pixel>" if opt_no_images==true, or replace them with dataURL asset strings, but please correct me if I'm wrong